server/src/main/java/org/elasticsearch/search/aggregations/support/ValuesSourceRegistry.java

```diff
@@ -151,13 +181,17 @@ public ValuesSourceRegistry build() {
     /** Maps Aggregation names to (ValuesSourceType, Supplier) pairs, keyed by ValuesSourceType */
     private final AggregationUsageService usageService;
     private Map<RegistryKey<?>, Map<ValuesSourceType, ?>> aggregatorRegistry;
+    private Map<String, ValuesSourceType> valueTypeLookup;
 
     public ValuesSourceRegistry(
         Map<RegistryKey<?>, List<Map.Entry<ValuesSourceType, ?>>> aggregatorRegistry,
-        AggregationUsageService usageService
+        AggregationUsageService usageService,
+        Map<String, ValuesSourceType> valueTypeLookup
     ) {
         this.aggregatorRegistry = copyMap(aggregatorRegistry);
         this.usageService = usageService;
+        // TODO: Make an immutable copy blah blah blah
```

Comment:
Is this planned for a follow-up?

Reply:
yeah, I skipped it for time when doing the initial draft, but I'll add that in before merging.  Thanks for the reminder.  Probably a few other small clean ups I need to do too (like fixing the merge conflicts...).